### ISSUE TITLE
Quotes around options with spaces should not be passed to jest

### ISSUE THREAD

**Issue by segrey** [meltedspark/angular-builders#69]
1. Generate a default Angular CLI project (for example, by running `npx --package @angular/cli ng new myNewProject` command).
2. Configure `ng test` to run jest according to https://www.npmjs.com/package/@angular-builders/jest
3. Run `ng test`, all tests are passed expectedly
4. Run single test:
```
ng test "--testNamePattern=^AppComponent should create the app$"
Test Suites: 1 skipped, 0 of 1 total
Tests:       3 skipped, 3 total
Snapshots:   0 total
Time:        1.206s, estimated 2s
Ran all test suites with tests matching ""^AppComponent should create the app$"".
```
All tests are skipped unexpectedly. Because jest sees `testNamePattern` value surrounded by double quote characters and constructs a wrong pattern therefore.

@angular-builders/jest: 1.2.2
jest: 23.6.0
@angular/cli: 6.2.3

**Comment by segrey** [comment]
Looks like it's caused by https://github.com/meltedspark/angular-builders/blob/c12ebf236a2e0834fdc4f99082e32706cfefbc43/packages/jest/src/options-converter.ts#L12
@meltedspark Is it possible to not enclose parameters with double quotes?

**Comment by meltedspark** [comment]
Wiuld you please try changing `"--testNamePattern=^AppComponent should create the app$"` to `--testNamePattern="^AppComponent should create the app$"`?

**Comment by meltedspark** [comment]
Interesting, seems to be some breaking change in Angular CLI. I tested this behavior with the exact same option (`--testNamePattern`) and it worked (that is the reason for quotes in `options-converter`, but now it is broken indeed. On it.

**Comment by segrey** [comment]
@meltedspark Thanks for looking into it! Sorry for the late response.
Not sure that @angular-builder/jest should quote any cli option passed to Jest as Jest expects (as any other CLI tool) that passed arguments are already processed, so quotes in argument are handled as literal quotes. For example, [process.argv](https://nodejs.org/docs/latest/api/process.html#process_process_argv) contains strings without quotes even if a node process is spawned in a shell with quoted arguments.

Regarding this issue, unfortunately, it's still reproduced with @angular-builders/jest@1.2.3.
See double quotes in `Ran all test suites with tests matching ""^AppComponent should create the app$""` in the following three examples:
```
$ ng test "--testNamePattern=^AppComponent should create the app$"

Test Suites: 2 skipped, 0 of 2 total
Tests:       6 skipped, 6 total
Snapshots:   0 total
Time:        1.504s, estimated 2s
Ran all test suites with tests matching ""^AppComponent should create the app$"".
```
```
$ ng test --testNamePattern="^AppComponent should create the app$"

Test Suites: 2 skipped, 0 of 2 total
Tests:       6 skipped, 6 total
Snapshots:   0 total
Time:        1.5s, estimated 2s
Ran all test suites with tests matching ""^AppComponent should create the app$"".
```
```
$ ng test --testNamePattern "^AppComponent should create the app$"

Test Suites: 2 skipped, 0 of 2 total
Tests:       6 skipped, 6 total
Snapshots:   0 total
Time:        1.528s, estimated 2s
Ran all test suites with tests matching ""^AppComponent should create the app$"".
```

When tests are running without shell, no quotes are needed:
```javascript
require('child_process').spawn('node', [
  'node_modules/@angular/cli/bin/ng',
  'test',
  '--testNamePattern',
  '^AppComponent should create the app$'
], {
  stdio: 'inherit'
});
```

Output:
```
Test Suites: 2 skipped, 0 of 2 total
Tests:       6 skipped, 6 total
Snapshots:   0 total
Time:        1.489s, estimated 2s
Ran all test suites with tests matching ""^AppComponent should create the app$"".
```

When Jest tests are running directly (using `jest` executable, not `ng`), no quotes for arguments are needed (it can be reproduced using https://github.com/facebook/jest/tree/master/examples/getting-started).

Are there cases when manually quoting is needed?

**Comment by meltedspark** [comment]
Hmm interesting observation. So you suggest there no case when I need to enclose the passed option with quotes?

**Comment by segrey** [comment]
Yes, it seems so, no quotes are needed here.  Quoting is needed when an array of arguments is converted to a single command line string. In this case quotes allow to decode arguments correctly from the command line string: that's what shell does when a command line typed manually is executed.

**Comment by meltedspark** [comment]
@segrey Can you confirm please the issue is fixed in `1.2.4`?

**Comment by segrey** [comment]
Confirming, the issue is fixed in `1.2.4`. Thanks!
